script.js

```diff
@@ -1,12 +1,126 @@
 //You can edit ALL of the code here
+
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  setUpSearch();
+  displayingAllEpisodesLength();
+  populateSelect(allEpisodes);
+  menu();
+}
+
+function setUpSearch() {
+  let searchInput = document.getElementById("search-input");
+  searchInput.addEventListener("keyup", filteredSearch);
+}
+
+function displayingAllEpisodesLength() {
+  let episodes = getAllEpisodes();
+  let display = document.querySelector(".search-results");
+  display.innerText = `Displaying ${episodes.length} / ${episodes.length} episodes`;
+}
+
+function filteredSearch() {
+  let searchInput = document.getElementById("search-input");
+  let searchInputValue = searchInput.value.toLowerCase();
+  let splitSearch = searchInputValue.split(" ");
+  // remove all new-div s from root
+  let root = document.getElementById("root");
+  document.querySelectorAll(".new-div").forEach((element) => element.remove());
+
+  // .filter on all episodes
+  let episodes = getAllEpisodes();
+  let filteredArr = episodes.filter((element) => {
+    let name = element.name.toLowerCase();
+    let paragraph = element.summary.toLowerCase();
+    let allText = name + " " + paragraph;
+    return containsAllSubStrings(allText, splitSearch);
+    // name.includes(searchInputValue) || paragraph.includes(searchInputValue)
+  });
+  // makepage for remaining episodes
+
+  function containsAllSubStrings(targetString, searchSubstrings) {
+    //loop over searchSubstrings and return true if all of them are included in targetString
+    return searchSubstrings.some((item) => targetString.includes(item));
+  }
```

Comment:
`.some` will return true if any item in `splitSearch` is in `allText` this will give you more results as you type more words.
If you replace `.some` with `.every` it will return true if every item is in `allText`